gunicorn/workers/async.py

```diff
@@ -53,10 +53,13 @@ def handle(self, listener, client, addr):
                 exc_info = sys.exc_info()
                 # pass to next try-except level
                 six.reraise(exc_info[0], exc_info[1], exc_info[2])
-            except socket.error:
-                exc_info = sys.exc_info()
-                # pass to next try-except level
-                six.reraise(exc_info[0], exc_info[1], exc_info[2])
+            except socket.error, err_info:
+                if err_info and isinstance(err_info, list) and err_info[0] == errno.EPIPE:
```

Comment:
err_info can be an exception message string or an array with errcode and exception message